python/lsst/ts/externalscripts/maintel/warmup_hexapod.py

```diff
@@ -0,0 +1,265 @@
+            )
+        )
+
+        # The estimated duration deppends on the configuration
+        self.activity_estimated_duration = None
+
+        # The checkpoints depend on the configuration
+        self.checkpoints_activities = [
+            ("Run warm-up sequence for hexapod", self.warm_up),
+        ]
+
+    @classmethod
+    def get_schema(cls):
+        yaml_schema = """
+            $schema: http://json-schema/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_standardscripts/maintel/WarmUpHexapod.yaml
+            title: WarmUpHexapod v1
+            description: Configuration for WarmUpHexapod.
+            type: object
+            properties:
+              hexapod:
+                description: Which hexapod will move? (camera_hexapod/m2_hexapod)
+                type: string
+                enum: ["camera_hexapod", "m2_hexapod"]
+              axis:
+                description: Which axis will move? (x, y, z, u, v, w).
+                type: string
+                enum: ["x", "y", "z", "u", "v", "w"]
+                default: z
+              step_size:
+                description: >-
+                  The discrete step size in microns in which the hexapod will move. 
+                  This can be a number or an array of numbers.
+                anyOf:
+                  - type: number
+                  - type: array
+                default: 250
+              sleep_time:
+                description: The sleep time in seconds between movements movements.
+                anyOf:
+                  - type: number
+                  - type: array
+                default: 1
+              max_position:
+                description: Maximum position (absolute value) for the movements.
+                type: number
+                default: 13000
+            additionalProperties: false
+            """
+        return yaml.safe_load(yaml_schema)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config: `types.SimpleNamespace`
+            Configuration data. See `get_schema` for information about data
+            structure.
+        """
+        self.log.debug(
+            f"Setting up configuration: \n"
+            f"  hexapod: {config.hexapod}\n"
+            f"  axis: {config.axis}\n"
+            f"  step_size: {config.step_size}\n"
+            f"  sleep_time: {config.sleep_time}\n"
+            f"  max_position: {config.max_position}\n"
+        )
+
+        self.config = config
+
+        # Converts the string to salindex
+        self.hexapod_name = config.hexapod
+        self.sal_index = getattr(SalIndex, self.hexapod_name.upper())
+        self.hexapod = getattr(self.mtcs.rem, f"mthexapod_{self.sal_index}")
+        self.log.debug(f"Using the {self.hexapod_name} with sal index {self.sal_index}")
+
+        # Assert step_size and sleep_time are compatible
+        if isinstance(self.config.step_size, (int, float)):
```

Comment:
It might be more reliable to check it `step_size` is iterable, e.g.;

```
from collections.abc import Iterable

if not isinstance(self.config.step_size, Iterable):
    ...
```